### PR TITLE
Use `is` for Locator handling

### PR DESCRIPTION
**Pull request by ghost**
IMO a bit cleaner, and allows other mods to have locator items by extending the Locator class

### CODE REVIEW

File: Components/StorageHeart.cs [hunk 1]
```diff
@@ -23,17 +23,17 @@ public override TEStorageHeart GetHeart(int i, int j)
 		public override bool RightClick(int i, int j)
 		{
 			Player player = Main.LocalPlayer;
-			Item item = player.HeldItem;
-			if (item.type == ModContent.ItemType<Locator>() || item.type == ModContent.ItemType<LocatorDisk>() || item.type == ModContent.ItemType<PortableAccess>())
+			ModItem item = player.HeldItem.ModItem;
```
Comment (ExterminatorX99) [inline Components/StorageHeart.cs]: Change this back to `Item item = player.HeldItem;`

File: Components/StorageHeart.cs [hunk 2]
```diff
@@ -23,17 +23,17 @@ public override TEStorageHeart GetHeart(int i, int j)
...
+
+			if (item != null && item is Locator locator)
```
Comment (ExterminatorX99) [inline Components/StorageHeart.cs]: Null check isn't needed, but a bit of extra safety can't hurt.
Change this to `if (!item.IsAir && item.ModItem is Locator locator)`

File: Components/StorageHeart.cs [hunk 3]
```diff
@@ -23,17 +23,17 @@ public override TEStorageHeart GetHeart(int i, int j)
...
 			{
 				if (Main.tile[i, j].TileFrameX % 36 == 18)
 					i--;
 				if (Main.tile[i, j].TileFrameY % 36 == 18)
 					j--;
-				Locator locator = (Locator) item.ModItem;
 				locator.location = new Point16(i, j);
 				if (player.selectedItem == 58)
-					Main.mouseItem = item.Clone();
+					Main.mouseItem = item.Item.Clone();
```
Comment (ExterminatorX99) [inline Components/StorageHeart.cs]: Earlier comment makes this not need changing

**Comment by ghost** [comment]
done :)

**Comment by absoluteAquarian** [comment]
`PortableCraftingAccess` inherits from `PortableAccess`.  Please add a `&& locator is not PortableCraftingAccess` condition to the check

**Comment by ghost** [comment]
> `PortableCraftingAccess` inherits from `PortableAccess`. Please add a `&& locator is not PortableCraftingAccess` condition to the check

sure, done

**Pull request merged**